IonDotnet/Internals/Text/RawTextReader.cs

```diff
@@ -374,6 +381,8 @@ private void ParseNext()
                         _state = StateEof;
                         return;
                     case ActionFinishContainer:
+                        if (_expectedContainerClosingSymbol.Count == 0 || _expectedContainerClosingSymbol.Pop() != token)
+                            throw new FormatException($"Illegal character");
```

Comment:
Let's make this error message more descriptive by saying which type of container the illegal character was encountered in and what the illegal character was.

Reply:
Done.
Now for an Ion like `` { test:1 ]`` the error would be:
``Illegal character: expected '}' character but encountered ']'``